ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsLocator.java

```diff
@@ -174,16 +182,23 @@
    */
   private List<PartitionMetadata>  getMetadataForAllPartitions(
       final Set<Integer> filterPartitions) {
+    final Set<String> sourceTopics = findSubtopologySourceTopics();
```

Comment:
If you choose to add some docs, it'd be worth explaining that the sources may have different numbers of partitions, but the store partitioning is determined by the _maximum_ number of partitions of _any_ source topic, which the algorithm mapping partition to host, below, achieves. It took me a minute to see that it's correct..

Reply:
I added a bunch of comments that aim to clarify this.